src/main/java/com/github/checkstyle/regression/module/ConfigGenerator.java

```diff
@@ -0,0 +1,64 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2017 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.github.checkstyle.regression.module;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.github.checkstyle.regression.data.ConfigNode;
+import com.github.checkstyle.regression.data.GitChange;
+import com.github.checkstyle.regression.data.ImmutableConfigNode;
+
+/**
+ * Contains the logic of regression config generation.
+ * @author LuoLiangchen
+ */
+public final class ConfigGenerator {
+    /** Prevents instantiation. */
+    private ConfigGenerator() {
+    }
+
+    /**
+     * Generates the regression config nodes from a list of changes.
+     * @param changes the changes source
+     * @return the regression config nodes generated from the given changes
+     */
+    public static List<ConfigNode> generateConfigNodes(List<GitChange> changes) {
+        final List<ModuleInfo> modules = changes.stream()
+                .filter(ModuleUtils::isCheckstyleModule)
+                .map(ModuleUtils::convertModuleChangeToModuleInfo)
+                .collect(Collectors.toList());
+        return modules.stream()
+                .map(module -> {
+                    final ConfigNode.Parent parent;
+                    if (module.hasCheckerParent()) {
+                        parent = ConfigNode.Parent.CHECKER;
```

Comment:
What value can `parent` be? Actually I felt there are only these two before... I also generate the json file by a simple if-else(If the module is a subclass of AbstractChecker, then TreeWalker, otherwise Checker)